### PR TITLE
chore: Enable fieldalignment go vet check

### PR DESCRIPTION
**Pull request by patriciareinoso**
Enable fieldalignment go vet check in CI
fieldalignment defines an Analyzer that detects structs that would use less memory if their fields were sorted.

### CODE REVIEW

File: context/context.go [hunk 1]
```diff
@@ -71,19 +67,22 @@ type SMFContext struct {
...
-	NrfCacheEvictionInterval time.Duration
 	StaticIpInfo             *[]factory.StaticIpInfo
+	CPNodeID                 pfcpType.NodeID
+	UDMProfile               models.NfProfile
+	NrfCacheEvictionInterval time.Duration
+	SBIPort                  int
+	LocalSEIDCount           uint64
+	EnableNrfCaching         bool
+	//*** For ULCL ** //
+	ULCLSupport bool
```
Comment (gab-arrobo) [inline context/context.go:85]:
```suggestion

	// For ULCL
	ULCLSupport bool
```

File: context/sm_context.go [hunk 2]
```diff
@@ -112,18 +110,16 @@ type SMContext struct {
...
-	AddUeLocation   *models.UserLocation `json:"addUeLocation,omitempty" yaml:"addUeLocation" bson:"addUeLocation,omitempty"` // ignore
-	OldPduSessionId int32                `json:"oldPduSessionId,omitempty" yaml:"oldPduSessionId" bson:"oldPduSessionId,omitempty"`
-	HoState         models.HoState       `json:"hoState,omitempty" yaml:"hoState" bson:"hoState,omitempty"`
+	AnType         models.AccessType    `json:"anType" yaml:"anType" bson:"anType"`
+	RatType        models.RatType       `json:"ratType,omitempty" yaml:"ratType" bson:"ratType,omitempty"`
+	PresenceInLadn models.PresenceState `json:"presenceInLadn,omitempty" yaml:"presenceInLadn" bson:"presenceInLadn,omitempty"` // ignore
+	UeLocation     *models.UserLocation `json:"ueLocation,omitempty" yaml:"ueLocation" bson:"ueLocation,omitempty"`
+	UeTimeZone     string               `json:"ueTimeZone,omitempty" yaml:"ueTimeZone" bson:"ueTimeZone,omitempty"`          // ignore
+	AddUeLocation  *models.UserLocation `json:"addUeLocation,omitempty" yaml:"addUeLocation" bson:"addUeLocation,omitempty"` // ignore
```
Comment (gab-arrobo) [inline context/sm_context.go:118]: What about grouping together the variables based on their type (when possible). for example, strings next to each other, pointers next to each other, etc.

File: factory/config.go [hunk 3]
```diff
@@ -79,29 +79,29 @@ type Configuration struct {
...
 
-	EnableNrfCaching         bool           `yaml:"enableNrfCaching"`
-	NrfCacheEvictionInterval int            `yaml:"nrfCacheEvictionInterval"`
 	StaticIpInfo             []StaticIpInfo `yaml:"staticIpInfo"`
+	NrfCacheEvictionInterval int            `yaml:"nrfCacheEvictionInterval"`
+	DebugProfilePort         int            `yaml:"debugProfilePort,omitempty"`
+	EnableNrfCaching         bool           `yaml:"enableNrfCaching"`
+	EnableDbStore            bool           `yaml:"enableDBStore,omitempty"`
+	EnableUpfAdapter         bool           `yaml:"enableUPFAdapter,omitempty"`
+	ULCL                     bool           `yaml:"ulcl,omitempty"`
```
Comment (gab-arrobo) [inline factory/config.go:93]: Why not keeping variables together based on their type (when possible)) such as below?
```diff
diff --git a/factory/config.go b/factory/config.go
index 7ca35d2..a1e803f 100644
--- a/factory/config.go
+++ b/factory/config.go
 type Configuration struct {
-       SmfName              string               `yaml:"smfName,omitempty"`
-       Sbi                  *Sbi                 `yaml:"sbi,omitempty"`
-       PFCP                 *PFCP                `yaml:"pfcp,omitempty"`
-       NrfUri               string               `yaml:"nrfUri,omitempty"`
-       UserPlaneInformation UserPlaneInformation `yaml:"userplane_information"`
-       ServiceNameList      []string             `yaml:"serviceNameList,omitempty"`
-       SNssaiInfo           []SnssaiInfoItem     `yaml:"snssaiInfos,omitempty"`
-       ULCL                 bool                 `yaml:"ulcl,omitempty"`
-       EnterpriseList       map[string]string    `yaml:"enterpriseList,omitempty"`
-       EnableDbStore        bool                 `yaml:"enableDBStore,omitempty"`
-       EnableUpfAdapter     bool                 `yaml:"enableUPFAdapter,omitempty"`
-       SmfDbName            string               `yaml:"smfDBName,omitempty"`
-       Mongodb              *Mongodb             `yaml:"mongodb,omitempty"`
-       KafkaInfo            KafkaInfo            `yaml:"kafkaInfo,omitempty"`
-       DebugProfilePort     int                  `yaml:"debugProfilePort,omitempty"`
-
-       EnableNrfCaching         bool           `yaml:"enableNrfCaching"`
-       NrfCacheEvictionInterval int            `yaml:"nrfCacheEvictionInterval"`
-       StaticIpInfo             []StaticIpInfo `yaml:"staticIpInfo"`
+       Mongodb                  *Mongodb             `yaml:"mongodb,omitempty"`
+       PFCP                     *PFCP                `yaml:"pfcp,omitempty"`
+       Sbi                      *Sbi                 `yaml:"sbi,omitempty"`
+       NrfUri                   string               `yaml:"nrfUri,omitempty"`
+       SmfName                  string               `yaml:"smfName,omitempty"`
+       SmfDbName                string               `yaml:"smfDBName,omitempty"`
+       SNssaiInfo               []SnssaiInfoItem     `yaml:"snssaiInfos,omitempty"`
+       StaticIpInfo             []StaticIpInfo       `yaml:"staticIpInfo"`
+       ServiceNameList          []string             `yaml:"serviceNameList,omitempty"`
+       EnterpriseList           map[string]string    `yaml:"enterpriseList,omitempty"`
+       KafkaInfo                KafkaInfo            `yaml:"kafkaInfo,omitempty"`
+       UserPlaneInformation     UserPlaneInformation `yaml:"userplane_information"`
+       DebugProfilePort         int                  `yaml:"debugProfilePort,omitempty"`
+       NrfCacheEvictionInterval int                  `yaml:"nrfCacheEvictionInterval"`
+       EnableDbStore            bool                 `yaml:"enableDBStore,omitempty"`
+       EnableNrfCaching         bool                 `yaml:"enableNrfCaching"`
+       EnableUpfAdapter         bool                 `yaml:"enableUPFAdapter,omitempty"`
+       ULCL                     bool                 `yaml:"ulcl,omitempty"`
 }
```

File: context/context.go (hunk 1 above)
Comment (patriciareinoso) [reply context/context.go]: done

File: context/sm_context.go (hunk 2 above)
Comment (patriciareinoso) [reply context/sm_context.go]: done

File: factory/config.go (hunk 3 above)
Comment (patriciareinoso) [reply factory/config.go]: done

**Review by gab-arrobo — APPROVED** [review]
+1 (tested changes with AiaB)

**Pull request merged**